tests/test_read_write/test_read_csv.py

```diff
@@ -0,0 +1,95 @@
+import pytest
+from itertools import product
+from clumper import Clumper
+
+
+paths = ["tests/monopoly.csv", "https://calmcode.io/datasets/monopoly.csv"]
+nrows = [(None, 22), (10, 10), (15, 15)]
+fields = [
+    (
+        None,
+        [
+            "name",
+            "rent",
+            "house_1",
+            "house_2",
+            "house_3",
+            "house_4",
+            "hotel",
+            "deed_cost",
+            "house_cost",
+            "color",
+            "tile",
+        ],
+        22,
+    ),
+    (
+        [
+            "namee",
+            "rent",
+            "house1",
+            "house2",
+            "house3",
+            "house4",
+            "hotell",
+            "deed_cost",
+            "house_cost",
+            "colour",
+            "tille",
+        ],
+        [
+            "namee",
+            "rent",
+            "house1",
+            "house2",
+            "house3",
+            "house4",
+            "hotell",
+            "deed_cost",
+            "house_cost",
+            "colour",
+            "tille",
+        ],
+        23,
+    ),
+]
+
+path_nrows = [(path, nrows, length) for path, (nrows, length) in product(paths, nrows)]
+path_fields = [
+    (path, fieldnames, fields_check, length)
+    for path, (fieldnames, fields_check, length) in product(paths, fields)
+]
+
+
+@pytest.mark.parametrize("path,nrows,length", path_nrows)
+def test_read_csv(path, nrows, length):
+    "Test that the length of clumper matches the total number of rows in the csv."
+    clump = Clumper.read_csv(path=path, nrows=nrows)
+    assert len(clump) == length
+
+
+@pytest.mark.parametrize("path,fieldnames,field_check,length", path_fields)
+def test_fieldnames(path, fieldnames, field_check, length):
+    "Test that fieldnames matches keys of Clumper. Also test number of rows returned."
+    clump = Clumper.read_csv(path=path, fieldnames=fieldnames)
+    assert not set(field_check).difference(clump.keys())
+    assert len(clump) == length
```

Comment:
Since the length is checked in `test_fieldnames` I'd be fine if this test limits itself to just checking the `fieldnames` argument.